### PR TITLE
checker: fix logic of enum value overflow check

### LINKED ISSUE #19315: Wrong enum overflow checks for the `int` type
### Describe the bug

Our code currently checks if enum values don't overflow the `int` type. The problem is that it throws an error for the value `2147483647`, a valid `int` value.

### Reproduction Steps

```
enum Test {
    a = 0
    b = 2147483647
}
```

### Expected Behavior

No error

### Current Behavior

```
code.v:3:6: error: enum value `2147483647` overflows the enum type `int`, values of which have to be in [-2147483648, 2147483647]
    1 | enum Test {
    2 |     a = 0
    3 |     b = 2147483647
      |         ~~~~~~~~~~
    4 | }
```

### Possible Solution

`2147483647` should be treated as valid `int` valuea value

### Additional Information/Context

Also, take into account that we also need to have a check for cases like this

```
enum Test {
    a = 0
    b = 2147483647
    c
}
```

The error should be rise

```
code.v:4:6: error: enum value `2147483647` overflows the enum type `int`, values of which have to be in [-2147483648, 2147483648]
    1 | enum Test {
    2 |     a = 0
    3 |     b = 2147483647
    4 |     c
      |     ~
    5 | }
```

### V version

V 0.4.1 b0fb9fb 

### Environment details (OS name and version, etc.)

Playground

### PR DESCRIPTION
**Pull request by ArtemkaKun**
Fix #19315 

This PR adds more mess to `checker.v` mess. It also fixes wrong logic (I have no idea why it was merged at all) in `atoi.v` - `parse_int`, but only for positive integers.

### <samp>🤖 Generated by Copilot at dad3bb7</samp>

This pull request improves the error handling and parsing of enum values in the V compiler. It updates the `common_parse_int` function to return a more informative error message, and the checker to use the `parse_int` and `parse_uint` methods for enum values. It also adds a new test file `enum_field_edge_values.vv` and removes an obsolete test file `enum_field_overflow.out`.

### <samp>🤖 Generated by Copilot at dad3bb7</samp>

*  Modify error message for integer overflow to include input string ([link](https://github.com/vlang/v/pull/19321/files?diff=unified&w=0#diff-5ab5e17885914d1c0e2aa0ad9b25a2ac673fecb717d9102681a9b729c698f762L196-R196))
*  Use parsing methods for enum values instead of direct integer conversion ([link](https://github.com/vlang/v/pull/19321/files?diff=unified&w=0#diff-22c1eb3704a15ee5526dec7b8cfe3f172f040b0aabfc682f2d76d55d174cac46L1790-R1853))
*  Add test for enums with edge values of underlying integer type ([link](https://github.com/vlang/v/pull/19321/files?diff=unified&w=0#diff-5aaaab7e4bd94150e64710a6f341f573e8c5baa654c15018531fbdc57ca6d7b2R1-R14))
*  Remove test for enum value overflow that is covered by new test ([link](https://github.com/vlang/v/pull/19321/files?diff=unified&w=0#diff-ef38cf7e65fe368a6e9eff6fbca8abf1184044c32833d0989710ae9cec24658fL1-L7))

### CODE REVIEW

**Comment by ArtemkaKun** [comment]
I have no idea why `native` stuff is failing, I need help there

**Comment by ArtemkaKun** [comment]
@Spydr06

File: vlib/v/checker/checker.v [hunk 1]
```diff
@@ -1787,33 +1787,69 @@ fn (mut c Checker) enum_decl(mut node ast.EnumDecl) {
 					mut ival := i64(0)
 
 					if signed {
-						val := field.expr.val.i64()
+						val := field.expr.val.parse_int(0, 64) or {
+							if err.msg().starts_with('common_parse_int: integer overflow') {
+								overflows = true
+								enum_imax
```
Comment (Delta456) [inline vlib/v/checker/checker.v:1793]: Is this the only way to do it?

Comment (ArtemkaKun) [reply vlib/v/checker/checker.v:1793]: This is what `strconv` exposes to me, so probably yes. Since there can be 4 different errors, I want to be sure that I handled the overflow error correctly

**Comment by ArtemkaKun** [comment]
Bruh, not gazillion tests failed because a flaw in `atoi` 👍 Probably we should fix it first

**Comment by Spydr06** [comment]
> I have no idea why `native` stuff is failing, I need help there

Your changes (?) seem to false-trigger on an int literal

> vlib/v/gen/native/amd64.v:3747:54: error: modulo by zero
> 3745 |             c.g.write8(0x45 + int(reg) % 8 * 8 + far_var_offset)
> 3746 |             if is_far_var {
> 3747 |                 c.g.write32(int((0xffffffff - i64(offset) + 1) % 0x100000000))
>      |                                                                  ~~~~~~~~~~~
> 3748 |             } else {
> 3749 |                 c.g.write8((0xff - offset + 1) % 0x100)

**Comment by JalonSolov** [comment]
That's not a false trigger, though... `0x100000000` is too large for 32-bit.

**Comment by spytheman** [comment]
> That's not a false trigger, though... 0x100000000 is too large for 32-bit.

Yes it is. It boils down to: `int( x % 0x100000000 )`, where `x` is i64. There should not be an error for that expression.

File: vlib/strconv/atoi.v [hunk 2]
```diff
@@ -193,8 +193,7 @@ pub fn common_parse_int(_s string, base int, _bit_size int, error_on_non_digit b
 	// TODO: check should u64(bit_size-1) be size of int (32)?
 	cutoff := u64(1) << u64(bit_size - 1)
 	if !neg && un >= cutoff {
-		// return error('parse_int: range error $s0')
-		return i64(cutoff - u64(1))
+		return error('common_parse_int: integer overflow ${s}')
 	}
```
Comment (spytheman) [inline vlib/strconv/atoi.v:197]: Make a separate PR just for that + a test for it.
